src/subscribe/channel.ts

```diff
@@ -135,12 +170,13 @@ export default class Channel<T extends Action = Action> {
     this.setInterval()
 
     try {
-      this.setResponse(await this.action(...args))
+      this.response = await this.action(...args)
       this.errored = false
       this.error = null
     } catch (error) {
       this.errored = true
       this.error = error
+      this.clearTimer()
```

Comment:
This makes sure the action isn't executed again if the action resulted in an error.